python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -771,6 +772,14 @@ def _config_type_for_pydantic_field(pydantic_field: ModelField) -> ConfigType:
 
 
 def _config_type_for_type_on_pydantic_field(potential_dagster_type: Any) -> ConfigType:
+    # special case pydantic constrained types to their source equivalents
+    if safe_is_subclass(potential_dagster_type, ConstrainedStr):
+        return StringSource
+    elif safe_is_subclass(potential_dagster_type, ConstrainedFloat):
+        potential_dagster_type = float
```

Comment:
I assume this is because we do not have FloatSource? If so comment to that effect.